### PR TITLE
Budget App - Full Website

### PR DESCRIPTION
**Pull request by aamirkhan2478**
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### Video Link: https://drive.google.com/file/d/1J3jDFveZznVjx5uZs9z9lN_yW7bPMWUU/view?usp=sharing

### CODE REVIEW

File: app/views/categories/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  <%= form_with model: @category, url:categories_path do |f| %>
+    <div class="form-group mb-3">
+      <%= f.select :icon, ["Select Category",["Housing","housing.png"], ["Transportation","transportation.png"], ["Food","burger.png"], ["Utilities","utilities.png"], ["Clothing","clothing.png"], ["Medical/Healthcare","medical.png"], ["Insurance","insurance.png"] ], {}, {class:"form-select"} %>
+    </div>
+    <div class="form-group mb-3">
+      <%= f.text_field :name,  class:"form-control", Placeholder:"Name"%>
+    </div>
+    <%= f.submit "Add Category",class:"btn btn-primary btn-sm" %>
+  </div>
+<% end %>
```
Comment (mirouhml) [inline app/views/categories/new.html.erb:13]: - [ ] Please know that it's required to have the option to add an icon while adding a new category, I don't see that option here:

  image

  Kindly add it as it is required.

File: app/views/transaction/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  <%= form_with model: @transaction, url:category_transaction_index_path(params[:category_id]) do |f| %>
+    <div class="form-group mb-3">
+      <%= f.text_field :name,  class:"form-control", Placeholder:"Name"%>
+    </div>
+    <div class="form-group mb-3">
+      <%= f.number_field :amount, class:"form-control", placeholder:"Amount" %>
+    </div>
+    <%= f.submit "Add Transaction",class:"btn btn-primary btn-sm" %>
+  </div>
+<% end %>
```
Comment (mirouhml) [inline app/views/transaction/new.html.erb:12]: - [ ] Please know that the new transaction form should have the following: 
  * name (mandatory)
  * amount (mandatory)
  * categories (mandatory at least one)
  
  What you're missing is the "categories" I know that you're adding the category automatically but you should have the option to select at least one.
  
  image
  
  Kindly add this as well as it is required!

File: app/views/shared/_navbar.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <% end %>
+    <a class="navbar-brand" href="#"><%= text %></a>
+    <% if current_user.present? %>
+      <%= button_to destroy_user_session_path, class:"btn", method: :delete do %>
+        <i class="fa fa-power-off text-danger"></i>
+      <% end %>
+    <% else %>
+      <i class="fa fa-user text-white"></i>
+    <% end %>
+  </div>
```
Comment (mirouhml) [inline app/views/shared/_navbar.html.erb:19]: - [ ] Please know that the back button for some reason took me back to the splash page as if I logged out, I recommend that you hide it entirely when you're on the home page so you're safe from these issues.

  image

File: app/views/transaction/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                <% credit += transaction.amount %>
+              <% end %>
+            <% end %>
+            <%= "$#{credit}" %>
+          </small>
+        </p>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (mirouhml) [inline app/views/transaction/index.html.erb:25]: - [ ] The icons are not visible for some reason on the deployed version, kindly look into that, this is how they look:

  
  image

**Review by mirouhml — CHANGES_REQUESTED** [review]
# REQUIRED CHANGES 🔴

Hi @aamirkhan2478, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @aamirkhan2478 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**